### PR TITLE
Add GetCertificateStats to StatsReport

### PR DESCRIPTION
**Pull request by obasajujoshua31**
Add an API to get basic stats around certificate

Relates to #610

#### Reference issue
Fixes #...

### CODE REVIEW

File: certificate.go [hunk 1]
```diff
@@ -155,5 +156,27 @@ func GenerateCertificate(secretKey crypto.PrivateKey) (*Certificate, error) {
...
+
+	fingerPrintAlgo, err := c.GetFingerprints()
+	if err != nil {
+		return err
+	}
+
+	stats := CertificateStats{
+		Timestamp:            statsTimestampFrom(time.Now()),
+		Type:                 StatsTypeCertificate,
+		ID:                   "certificate",
```
Comment (tarrencev) [inline certificate.go:173]:
```suggestion
		ID:                   c.statsID,
```

File: certificate.go [hunk 2]
```diff
@@ -155,5 +156,27 @@ func GenerateCertificate(secretKey crypto.PrivateKey) (*Certificate, error) {
...
+		Fingerprint:          fingerPrintAlgo[0].Value,
+		FingerprintAlgorithm: fingerPrintAlgo[0].Algorithm,
+		Base64Certificate:    string(c.x509Cert.Raw),
```
Comment (tarrencev) [inline certificate.go:176]: Is this base64 encoded?

File: certificate.go [hunk 3]
```diff
@@ -155,5 +156,27 @@ func GenerateCertificate(secretKey crypto.PrivateKey) (*Certificate, error) {
...
+		IssuerCertificateID:  "",
```
Comment (tarrencev) [inline certificate.go:177]: Issuer should be set. You can retrieve it from `c.X509Cert.Issuer`

File: peerconnection.go [hunk 4]
```diff
@@ -1947,9 +1947,18 @@ func (pc *PeerConnection) GetStats() StatsReport {
...
-	pc.mu.Unlock()
 
 	statsCollector.Collect(stats.ID, stats)
+
+	certificates := pc.configuration.Certificates
+	for _, certificate := range certificates {
+		err := certificate.collectStats(statsCollector)
+		if err != nil {
+			continue
+		}
```
Comment (tarrencev) [inline peerconnection.go:1958]:
```suggestion
		if err := certificate.collectStats(statsCollector); err != nil {
		        continue
		}
```

**Review by tarrencev — CHANGES_REQUESTED** [review]

File: stats_go.go [hunk 5]
```diff
@@ -61,3 +61,18 @@ func (r StatsReport) GetICECandidatePairStats(c *ICECandidatePair) (ICECandidate
...
 	return candidateStats, true
 }
+
+// GetCertificateStats is a helper method to return the associated stats for a given Certificate
+func (r StatsReport) GetCertificateStats(c *Certificate) (CertificateStats, bool) {
+	statsID := c.statsID
+	stats, ok := r[statsID]
+	if !ok {
+		return CertificateStats{}, false
+	}
```
Comment (tarrencev) [inline stats_go.go:71]:
```suggestion
	if stats, ok := r[statsID];  !ok {
		return CertificateStats{}, false
	}
```

**Comment by tarrencev** [comment]
@obasajujoshua31 if you can, add a simple test to get codecov/patch to pass. otherwise this looks good!

File: certificate.go [hunk 6]
```diff
@@ -155,5 +157,26 @@ func GenerateCertificate(secretKey crypto.PrivateKey) (*Certificate, error) {
...
 func CertificateFromX509(privateKey crypto.PrivateKey, certificate *x509.Certificate) Certificate {
-	return Certificate{privateKey, certificate}
+	return Certificate{privateKey, certificate, fmt.Sprintf("certificate-%d", time.Now().UnixNano())}
+}
+
+func (c Certificate) collectStats(report *statsReportCollector) {
+	report.Collecting()
+
+	// Unable to cause it to error
+	fingerPrintAlgo, _ := c.GetFingerprints()
```
Comment (tarrencev) [inline certificate.go:167]: we should handle the err case here

**Review by Sean-Der — APPROVED** [review]
Really great work @obasajujoshua31 

Also a huge :+1: for that review @tarrencev, something we should all aspire too!

**Comment by tarrencev** [comment]
awesome!

**Pull request merged**